File: BasicMaths/AmstrongNumber/ArmstrongNumber.ts
class ArmstrongNumber {
  isArmstrongNumber(n: number): void {
    if (n == 0) console.log(true);
    else {
      let duplicate: number = n;
      let sum: number = 0;
      let digits: number = Math.floor(Math.log10(n) + 1);
      while (n > 0) {
        sum += Math.floor(Math.pow(n % 10, digits));
        n = Math.floor(n / 10);
      }
      console.log(duplicate == sum);
    }
  }
}

const armstrongNumber = new ArmstrongNumber();
armstrongNumber.isArmstrongNumber(9474);
armstrongNumber.isArmstrongNumber(1234);

File: BasicMaths/AmstrongNumber/ArmstrongNumber.py
import math
class ArmstrongNumber:
    def isArmstrongNumber(self, number):
        if number == 0:
            print(True)
        else:
            duplicate = number
            digits = int(math.log10(number) + 1)
            sum = 0
            while(number > 0):
                sum += int(math.pow(number % 10, digits))
                number //= 10
            print(duplicate == sum)
    
    def run(self):
        n = int(input("Enter the number: "))
        self.isArmstrongNumber(n)
armstrongNumber = ArmstrongNumber()
armstrongNumber.run()

